compiler/src/dotty/tools/dotc/transform/SyntheticMembers.scala

```diff
@@ -377,7 +384,13 @@ class SyntheticMembers(thisPhase: DenotTransformer) {
    *  O is O.type.
    */
   def ordinalBody(cls: Symbol, param: Tree)(implicit ctx: Context): Tree =
-    if (cls.is(Enum)) param.select(nme.enumTag)
+    if (cls.is(Enum)) {
+      val ordinalMeth = param.select(nme.ordinal)
+      val derivesFromJEnum =
```

Comment:
I wonder whether you can do just
```
param.select(nme.ordinal).ensureApplied()
```
That would mean you need no derivesFromJEnum test. If you do need the test, it's better to factor it out from here and CompleteJanaEnums and put it in SymUtils.